torch/fx/subgraph_rewriter.py

```diff
@@ -0,0 +1,308 @@
+                        nodes_to_delete.remove(subgraph_output)
+                        for arg in delete_args:
+                            original_graph.erase_node(arg)
+                            nodes_to_delete.remove(arg)
+                elif n.op == "placeholder":
+                    # If `n` corresponds to a placeholder node that
+                    # already exists in `original_graph`, we don't want
+                    # to swap it out
+                    if n.target in original_graph_inputs.keys():
+                        original_graph_node = original_graph_inputs[n.target]
+                        if original_graph_node in nodes_to_delete:
+                            nodes_to_delete.remove(original_graph_node)
+                        copied_nodes[n] = original_graph_node
+                    else:
+                        # Temporarily change the insertion point to the
+                        # beginning of the graph to ensure that placeholder
+                        # nodes are inserted before anything else
+                        with original_graph.inserting_after(last_placeholder):
+                            copied_nodes[n] = original_graph.node_copy(n,
+                                                                       lambda n :
+                                                                       copied_nodes[n])
+                            if n.name in original_graph_inputs.keys():
+                                original_node = original_graph_inputs[n.name]
+                                nodes_to_delete.add(original_node)
+                                original_node.replace_all_uses_with(copied_nodes[n])
+                            last_placeholder = copied_nodes[n]
+                else:
+                    copied_nodes[n] = original_graph.node_copy(n,
+                                                               lambda n :
+                                                               copied_nodes[n])
+
+        def erase_nodes(n : Node) -> None:
+            users = n.all_input_nodes
+            if n in nodes_to_delete:
+                original_graph.erase_node(n)
+            for user in users:
+                erase_nodes(user)
+
+        original_graph_output = next(iter(reversed(original_graph.nodes)))
+        erase_nodes(original_graph_output)
+
+        # During graph replacement, we don't add `replacement`
+        # placeholder nodes if a node by the same name already exists
+        # in `original_graph`. This means that the we need to manually
+        # update the placeholder nodes' "users" list to reflect the
+        # current state of the graph. By updating placeholders from the
+        # nodes that use them, we ensure that we have the correct
+        # use-def relationships.
+        for n in original_graph.nodes:
+            for i in n.all_input_nodes:
+                if i.op == "placeholder":
+                    i.users[n] = None
```

Comment:
This shouldn't be necessary so long as args/kwargs of the users actually points to the right node